python/meterpreter/ext_server_stdapi.py

```diff
@@ -2755,6 +2762,106 @@ def stdapi_ui_get_idle_time(request, response):
     response += tlv_pack(TLV_TYPE_IDLE_TIME, idle_time)
     return ERROR_SUCCESS, response
 
+@register_function_if(has_windll)
+def stdapi_ui_desktop_enum(request, response):
+    global ui_desktop_enum_RESPONSE
+    ui_desktop_enum_RESPONSE = response
+
+    class EnumDesktops_Param(ctypes.Structure):
+        _fields_ = [
+            ('sessionId', ctypes.c_ulong),
+            ('cpStationName', ctypes.c_char_p)
+        ]
+
+    EnumDesktopCallbackPrototype = ctypes.WINFUNCTYPE(ctypes.c_long, ctypes.c_char_p, ctypes.POINTER(EnumDesktops_Param))
+    EnumDesktopsA = ctypes.windll.user32.EnumDesktopsA
+    EnumDesktopsA.argtypes = [ctypes.c_void_p, EnumDesktopCallbackPrototype, ctypes.POINTER(EnumDesktops_Param)]
+    EnumDesktopsA.restype = ctypes.c_long
+
+    WindowStationCallbackPrototype = ctypes.WINFUNCTYPE(ctypes.c_long, ctypes.c_char_p, ctypes.c_ulong)
+    EnumWindowStationsA = ctypes.windll.user32.EnumWindowStationsA
+    EnumWindowStationsA.argtypes = [WindowStationCallbackPrototype, ctypes.c_ulong]
```

Comment:
The second parameter here should be `LPARAM`. That'll be a different size than `c_ulong` on 64-bit systems.

https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-enumwindowstationsa
https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types

Reply:
Should I define LPARAM like this?
```
if ctypes.sizeof(ctypes.c_long) == ctypes.sizeof(ctypes.c_void_p):
    WPARAM = ctypes.c_ulong
    LPARAM = ctypes.c_long
elif ctypes.sizeof(ctypes.c_longlong) == ctypes.sizeof(ctypes.c_void_p):
    WPARAM = ctypes.c_ulonglong
    LPARAM = ctypes.c_longlong
```
Same as definition in `wintypes`. Global or in `stdapi_ui_desktop_enum` function scope?

Reply:
For `LPARAM`, per the `LONG_PTR` definition in https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types
If you're on a 64-bit system it should be `ctypes.c_int64` and if you're not it should be `ctypes.c_long`.

Reply:
I meant definition scope and is that ok to redefine the block from `wintypes` or not. review changes in new commit.